app/controllers/chat_channel_memberships_controller.rb

```diff
@@ -25,15 +25,17 @@ def chat_channel_info
     @membership = ChatChannelMembership.find(params[:id])
     authorize @membership
     @channel = @membership.chat_channel
-    data = ChatChannelDetailPresenter.new(@channel, @membership).as_json
-
-    render json: { success: true, result: data, message: "" }, success: :ok
+    invite_cache_key = "chat-channel-invite-#{@channel.id}"
+    invitation_slug = Rails.cache.fetch(invite_cache_key, expires_in: 12.hours) do
+      "invitation-link-#{SecureRandom.hex(3)}"
+    end
+    @invitation_link = "/join_channel_invitation/#{@channel.slug}?invitation_slug=#{invitation_slug}"
```

Comment:
We need to add `invitation_slug` to our [approved list of Fastly params](https://github.com/forem/forem/blob/master/config/fastly/snippets/safe_params_list.vcl#L5) otherwise it will get stripped and this wont work.